src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -44,11 +51,19 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             {
                 logger.Error(MsalErrorMessage.ClientCredentialWrongAuthority);
             }
-
             AuthenticationResult authResult;
 
-            // Skip checking cache when force refresh or claims are specified
-            if (_clientParameters.ForceRefresh || !string.IsNullOrEmpty(AuthenticationRequestParameters.Claims))
+            // Skip cache if either:
+            // 1) ForceRefresh is set, or
+            // 2) Claims are specified and there is no AccessTokenHashToRefresh.
```

Comment:
Shouldn't you check here to see if forceRefresh and AccessTokenHashToRefresh are both provided and throw an error if so?

Reply:
I see you're checking ^ in Validate inside of AcquireTokenForClientParameterBuilder. Does that check happen before or after this code?

Reply:
If after, should it happen here instead?

Reply:
before